### PR TITLE
Support '\r' and '\r\n' line endings, closes scala/bug#5669

### PR DESCRIPTION
**Pull request by Philippus**

### CODE REVIEW

File: shared/src/main/scala/scala/util/parsing/input/OffsetPosition.scala [hunk 1]
```diff
@@ -34,10 +34,14 @@ case class OffsetPosition(source: CharSequence, offset: Int) extends Position {
 
   private def genIndex: Array[Int] = {
     val lineStarts = new ArrayBuffer[Int]
-    lineStarts += 0
-    for (i <- 0 until source.length)
-      if (source.charAt(i) == '\n') lineStarts += (i + 1)
-    lineStarts += source.length
+    lineStarts += 0 // first line
+    for (i <- 0 until source.length) {
```
Comment (javax-swing) [inline shared/src/main/scala/scala/util/parsing/input/OffsetPosition.scala]: may as well use `for i <- 1 until source.length` and remove the `if i>= 1` checks below

File: shared/src/main/scala/scala/util/parsing/input/OffsetPosition.scala [hunk 2]
```diff
@@ -34,10 +34,14 @@ case class OffsetPosition(source: CharSequence, offset: Int) extends Position {
...
+      if (i >= 1 && source.charAt(i - 1) == '\n') // \n or \r\n
```
Comment (javax-swing) [inline shared/src/main/scala/scala/util/parsing/input/OffsetPosition.scala]: the majority of the time the last character won't be a `\r` or `\n` might be worth binding `charAt(i - 1)` to a variable to save the extra lookup (dependant on the performance requirements of this code)

**Comment by SethTisue** [comment]
in the Scala community build, this build caused this Twirl test case to start failing: https://github.com/playframework/twirl/blob/68ad7b378be54887a09c5a67792bacc933d17d68/parser/src/test/scala/play/twirl/parser/test/ParserSpec.scala#L187-L189 (referencing https://github.com/playframework/twirl/blob/master/parser/src/test/resources/unclosedBracket2.scala.html)

```
[play-twirl] [info]   x unclosedBracket2.scala.html
[play-twirl] [error]      '31' is not equal to '32' (ParserSpec.scala:37)
```

the test file doesn't have any `\r` or `\r\n` line endings, so it seems probable it's a real regression? wdyt?

**Comment by Philippus** [comment]
I've reproduced it:
contents of the index generated by `genIndex` differs:
new: `ArrayBuffer(0, 85, 86, 111, 112, 136, 155, 166, 167, 169, 170, 185, 186, 207, 208, 233, 234, 243, 265, 266, 291, 292, 310, 335, 336, 346, 347, 356, 357, 367, 368, 370)`
vs
old: `ArrayBuffer(0, 85, 86, 111, 112, 136, 155, 166, 167, 169, 170, 185, 186, 207, 208, 233, 234, 243, 265, 266, 291, 292, 310, 335, 336, 346, 347, 356, 357, 367, 368, 370, 370)`
will work on a fix.

**Pull request merged**